### PR TITLE
Feature/implement command line tool

### PR DESCRIPTION
**Pull request by TorstenGoerg**

### CODE REVIEW

File: plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandLineHandler.java [hunk 1]
```diff
@@ -24,104 +25,71 @@ public CommandLineHandler()
...
 		}
+
+		String[] francaFilePaths = parsedArguments.getOptionValues('f');
+		String[] araFilePaths = parsedArguments.getOptionValues('a');
+//		// We expect at least one fidl/fdepl file as command line argument
+//		if (files.size() > 0 && files.get(0) != null) {
+		if ((francaFilePaths == null || francaFilePaths.length == 0) && (araFilePaths == null || araFilePaths.length == 0)) {
+			ConsoleLogger.logError("At least one input model file has to be given!");			
+		}		
+		if ((francaFilePaths != null && francaFilePaths.length > 0) && (araFilePaths != null && araFilePaths.length > 0)) {
```
Comment (kbirken) [inline plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandLineHandler.java]: Introducing two boolean local vars "haveFrancaInput" and "haveARAInput" would remove redundancy from lines 59 and 62.

File: plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandlineToolMain.java [hunk 2]
```diff
@@ -217,103 +310,56 @@ protected String normalize(String _path) {
 	public void setNoCommonCode() {
 		preferences.setPreference(PreferencesConstants.P_GENERATE_COMMON_DBUS,
 				"false");
-		ConsoleLogger.printLog("No common code will be generated");
+		ConsoleLogger.logInfo("No common code will be generated");
```
Comment (kbirken) [inline plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandlineToolMain.java]: This is probably not needed, no code is generated anyway.

File: plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandlineToolMain.java [hunk 3]
```diff
@@ -217,103 +310,56 @@ protected String normalize(String _path) {
...
 	}
 
 	public void setNoProxyCode() {
 		preferences.setPreference(PreferencesConstants.P_GENERATE_PROXY_DBUS,
 				"false");
-		ConsoleLogger.printLog("No proxy code will be generated");
+		ConsoleLogger.logInfo("No proxy code will be generated");
```
Comment (kbirken) [inline plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandlineToolMain.java]: This is probably not needed, no code is generated anyway.

File: plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandlineToolMain.java [hunk 4]
```diff
@@ -217,103 +310,56 @@ protected String normalize(String _path) {
...
 	}
 
 	public void setNoStubCode() {
 		preferences.setPreference(PreferencesConstants.P_GENERATE_STUB_DBUS,
 				"false");
-		ConsoleLogger.printLog("No stub code will be generated");
+		ConsoleLogger.logInfo("No stub code will be generated");
```
Comment (kbirken) [inline plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandlineToolMain.java]: This is probably not needed, no code is generated anyway.

File: plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARAPrimitveTypesCreator.xtend [hunk 5]
```diff
@@ -16,18 +16,16 @@ class ARAPrimitveTypesCreator extends Franca2ARABase {
 
 	private static final Logger logger = Logger.getLogger(ARAPrimitveTypesCreator.name)
 	
-	//todo: use a more reliable path
-//	val static String PATH_TO_STD_ARXML_FILES = "../../plugins/org.genivi.faracon/models/stdtypes.arxml"
-	val static String PATH_TO_STD_ARXML_FILES = "C:/Users/tgoerg/git/franca_ara_tools/plugins/org.genivi.faracon/models/stdtypes.arxml"
+	val static String PATH_TO_STD_ARXML_FILE = "stdtypes.arxml"
```
Comment (kbirken) [inline plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARAPrimitveTypesCreator.xtend]: This is defined also in ARAConnector.java. One of these redundant definitions should probably be avoided.

File: plugins/org.genivi.faracon/src/org/genivi/faracon/preferences/PreferencesConstants.java [hunk 6]
```diff
@@ -8,17 +8,14 @@
...
-    public static final String P_OUTPUT_STUBS     		= "outputDirStubs";
-	public static final String P_OUTPUT_COMMON    		= "outputDirCommon";
-	public static final String P_OUTPUT_SKELETON  		= "outputDirSkeleton";
-	public static final String P_OUTPUT_DEFAULT   		= "outputDirDefault";
-	public static final String P_OUTPUT_SUBDIRS   		= "outputSubdirs";
+    public static final String P_OUTPUT_DIRECTORY_PATH	= "outputDirectoryPath";
 	public static final String P_GENERATE_COMMON  		= "generatecommon";
     public static final String P_GENERATE_PROXY    		= "generateproxy";
     public static final String P_GENERATE_STUB     		= "generatestub";
     public static final String P_GENERATE_SKELETON 		= "generateskeleton";
```
Comment (kbirken) [inline plugins/org.genivi.faracon/src/org/genivi/faracon/preferences/PreferencesConstants.java]: I guess the P_GENERATE_... options are not needed for faracon and could be removed?

File: plugins/org.genivi.faracon/src/org/genivi/faracon/preferences/PreferencesConstants.java [hunk 7]
```diff
@@ -8,17 +8,14 @@
...
-	public static final String P_LOGOUTPUT        		= "logoutput";
+	public static final String P_LOGOUTPUT        		= "logOutput";
+	public static final String P_WARNINGS_AS_ERRORS 	= "warningsAsErrors";
+	public static final String P_CONTINUE_ON_ERRORS 	= "continueOnErrors";
 	public static final String P_ENUMPREFIX       		= "enumprefix";
 	public static final String P_SKELETONPOSTFIX  		= "skeletonpostfix";
```
Comment (kbirken) [inline plugins/org.genivi.faracon/src/org/genivi/faracon/preferences/PreferencesConstants.java]: The P_SKELETONPOSTFIX option is probably not needed?

File: plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARAPrimitveTypesCreator.xtend (hunk 5 above)
Comment (TorstenGoerg) [reply plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARAPrimitveTypesCreator.xtend]: This problem goes a little bit deeper. At both locations the ARXML that contains definitions of standard primitive types is loaded although these definitons are generally necessary for each AUTOSAR model (for boch directions, loading and storing). A better solution might be a special resource set class for AUTOSAR models which implicitly loads the standard defintions.

Comment (kbirken) [reply plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARAPrimitveTypesCreator.xtend]: So maybe we should tackle this at a later stage (we planned to build a configurable solution for AUTOSAR stdtypes anyway, e.g. using an extension mechanism). So probably keep it for now and create another issue so we won't forget this?

Comment (TorstenGoerg) [reply plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARAPrimitveTypesCreator.xtend]: I have just solved the problem by extracting the handling of standard primitive types into a specialized resource set class, as suggested above. Furthermore, this new resource set class will be a proper place to add configurability for AUTOSAR standard types at a later stage.

**Pull request merged**